nifi-extension-bundles/nifi-slack-bundle/nifi-slack-processors/src/main/java/org/apache/nifi/processors/slack/GetSlackReaction.java

```diff
@@ -0,0 +1,294 @@
+        @WritesAttribute(attribute = GetSlackReaction.ATTR_MESSAGE_TIMESTAMP, description = "The timestamp of the Slack message the reactions are fetched for."),
+        @WritesAttribute(attribute = "slack.reaction.<emoji name>", description = "The name of the emoji and the reaction count that was provided for the message."),
+        @WritesAttribute(attribute = GetSlackReaction.ATTR_WAIT_TIME, description = "The total number of minutes waited while the reactions were captured."),
+        @WritesAttribute(attribute = GetSlackReaction.ATTR_ERROR_MESSAGE, description = "The error message on fetching reactions.")
+})
+@SeeAlso({ListenSlack.class, ConsumeSlack.class, PublishSlack.class})
+@Tags({"slack", "conversation", "reactions.get", "social media", "emoji"})
+@CapabilityDescription("Retrieves reactions for a given message. The reactions are written as attributes.")
+@DefaultSettings(penaltyDuration = "5 min")
+public class GetSlackReaction extends AbstractProcessor {
+    public static final String ATTR_WAIT_TIME = "minutes.waited";
+    public static final String ATTR_ERROR_MESSAGE = "error.message";
+    public static final String ATTR_MESSAGE_TIMESTAMP = "slack.message.timestamp";
+    public static final String ATTR_CHANNEL_ID = "slack.channel.id";
+    public static final int PENALTY_MINUTES = 5;
+
+    static final PropertyDescriptor ACCESS_TOKEN = new PropertyDescriptor.Builder()
+            .name("Access Token")
+            .description("OAuth Access Token used for authenticating/authorizing the Slack request sent by NiFi. This may be either a User Token or a Bot Token. " +
+                    "It must be granted the reactions:read scope and channels:history, groups:history, im:history or mpim:history, depending on the type of conversation being used.")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .required(true)
+            .sensitive(true)
+            .build();
+
+    static final PropertyDescriptor CHANNEL_ID = new PropertyDescriptor.Builder()
+            .name("Channel ID")
+            .description("The ID of the channel.")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .build();
+
+    static final PropertyDescriptor THREAD_TIMESTAMP = new PropertyDescriptor.Builder()
+            .name("Message Timestamp")
+            .description("Slack message's timestamp the reactions are fetched for.")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .build();
+
+    static final PropertyDescriptor RELEASE_IF_ONE_REACTION = new PropertyDescriptor.Builder()
```

Comment:
RELEASE_PER_REACTION is probably a clearer naming

Reply:
@joewitt Based on the description change I got a bit confused. I'd like to clarify if the original intent of this property was clear enough. 
The original idea was to wait for the first reaction only, stop waiting for further reactions if this property is 'true' (even if wait perios is not elapsed yet). 
I am not sure if you also meant this or you'd like to release a flow file each time a new reactions arrives (till the wait period is elapsed).

Reply:
It is clear.  I still think 'release_per_reaction' makes more sense as to what the user means.  Basically what we mean to say is "Release as soon as we get any reaction" but that sounds too long.  

overall though I'm hesitant to be in code review mode based on the initial description and pr. You might want to consider starting with a more descriptive JIRA and then a PR that aligns to it.